applications/ShapeOptimizationApplication/python_scripts/custom_ios/wrl_reader.py

```diff
@@ -0,0 +1,98 @@
+# ==============================================================================
+#  KratosShapeOptimizationApplication
+#
+#  License:         BSD License
+#                   license: ShapeOptimizationApplication/license.txt
+#
+#  Main authors:    Geiser Armin, https://github.com/armingeiser
+#
+# ==============================================================================
+
+class Shape:
+    def __init__(self):
+        self.name = None
+        self.nodes = []
+        self.triangles = []
+
+
+def read_nodes(line, file):
+    nodes = []
+    while True:
+        if "[" in line:
+            break
+
+    line = next(file)
+
+    while True:
+        if "]" in line:
+            break
+        elif line == "\n":
+            line = next(file)
+
+        try:
+            entries = line.split(",")[0].split()
+            if len(entries) != 3:
+                raise RuntimeError("Did not find 3 coordinate components!", line)
```

Comment:
I would prepend "wrl_reader: Did not find 3 ...." to let the user know where it failed. This also applies to the remaining error messages in this file.